docs/fastdds/rtps_layer/rtps_layer.rst

```diff
@@ -174,3 +189,102 @@ You can specify a maximum amount of changes for the History to hold and an initi
 When the initial amount of reserved changes is lower than the maximum, the History will allocate more changes as they
 are needed until it reaches the maximum size.
 
+Using a custom Payload Pool
+---------------------------
+
+A *Payload* is defined as the data the user wants to transmit between a Writer and a Reader.
+RTPS needs to add some metadata to this Payload in order to manage the communication between the endpoints.
+Therefore, this Payload is encapsulated inside the :cpp:member:`SerializedPayload_t` field of the :class:`CacheChange_t`,
+while the rest of the fields of the :class:`CacheChange_t` provide the required metadata.
+
+|WriterHistory-api| and |ReaderHistory-api| provide an interface for the user to interact with these changes:
+Changes to be transmited by the Writeres are added to its WriterHistory,
+and changes already processed on the Reader can be removed from the ReaderHistory.
+In this sense, the History acts as a buffer for changes that are not fully processed yet.
+
+During a normal execution, new changes are added to the History and old ones are removed from it.
+In order to manage the lifecycle of the Payloads contained in these changes,
+Readers and Writers use a pool object,
+an implementation of the |IPayloadPool-api| interface.
+Different pool implementations allow for different optimizations.
+For example, Payloads of different size could be retrieved from different preallocated memory chunks.
+
+Writers and Readers can automatically select a default Payload pool implementation that best suits
+the configuration given in |HistoryAttributes-api|.
+However, a custom Payload pool can be given to |RTPSDomain::createRTPSWriter-api| and
+|RTPSDomain::createRTPSReader-api| functions.
+Writers and Readers will use the provided pool when a new :class:`CacheChange_t` is requested
+or released.
+
```

Comment:
May be interesting to add a note regarding the requirements for using a custom pool in the context of standard RTPS serialization. Mentioning section 10.2 of the RTPS standard and the 4 bytes required for `representation_identifier` and `representation_options`.